### PR TITLE
Feature/implement get all method filter

### PR DESCRIPTION
**Pull request by kostiantyn-lysenko**

### CODE REVIEW

File: internal/app/api/controller/event.go [hunk 1]
```diff
@@ -1,25 +1,123 @@
...
+	"strconv"
+	"workshop2/internal/app/errs"
+	"workshop2/internal/app/models"
+
+	"github.com/gorilla/mux"
 )
 
-func GetAllEvents(w http.ResponseWriter, r *http.Request) {
+type EventServiceInterface interface {
+	GetAll(interval string) []models.Event
```
Comment (ozakernychnyi) [inline internal/app/api/controller/event.go]: this method should return **error**

File: internal/app/api/controller/event.go [hunk 2]
```diff
@@ -1,25 +1,123 @@
...
+	Get(id int) (models.Event, error)
+	Create(event models.Event) models.Event
```
Comment (ozakernychnyi) [inline internal/app/api/controller/event.go]: this method should return **error** also

File: internal/app/api/controller/notification.go [hunk 3]
```diff
@@ -1,17 +1,78 @@
...
+	"workshop2/internal/app/errs"
+	"workshop2/internal/app/models"
+
+	"github.com/gorilla/mux"
 )
 
-func GetAllNotifications(w http.ResponseWriter, r *http.Request) {
+type NotificationServiceInterface interface {
+	GetAll(interval string) []models.Notification
+	Create(notification models.Notification) models.Notification
```
Comment (ozakernychnyi) [inline internal/app/api/controller/notification.go]: both methods should be able to return error

File: internal/app/repositories/event.go [hunk 4]
```diff
@@ -1,86 +1,79 @@
...
 	var event models.Event
-	var found bool
 
-	storage.DB.Lock()
-	for _, e := range storage.DB.Events {
+	r.RLock()
+	defer r.RUnlock()
+	for _, e := range r.Events {
 		if e.ID == id {
 			event = e
```
Comment (ozakernychnyi) [inline internal/app/repositories/event.go]: this step is redundant, just
`return e, nil`

File: internal/app/repositories/event.go [hunk 5]
```diff
@@ -1,86 +1,79 @@
...
-			found = true
-			break
+			return event, nil
 		}
 	}
-	storage.DB.Unlock()
 
-	return &event, found
+	return event, &errs.EventNotFoundError{}
```
Comment (ozakernychnyi) [inline internal/app/repositories/event.go]: you can return here
`return models.Event{}, &errs.EventNotFoundError{}`
and not to create `var event models.Event` in 21 line

File: internal/app/repositories/event.go [hunk 6]
```diff
@@ -1,86 +1,79 @@
...
 }
 
-func (r *EventRepository) Create(event *models.Event) *[]models.Event {
-	storage.DB.Lock()
-	id := len(storage.DB.Events) + 1
-	storage.DB.Unlock()
+func (r *EventRepository) Create(event models.Event) models.Event {
+	r.RLock()
```
Comment (ozakernychnyi) [inline internal/app/repositories/event.go]: it is not good idea to use RLock here, you can lose updated length of your Events slice, so better start this method with
```
r.Lock()
defer r.Unlock()
```

File: internal/app/repositories/notification.go [hunk 7]
```diff
@@ -1,66 +1,51 @@
...
-
-	storage.DB.Lock()
-	for _, e := range storage.DB.Notifications {
-		if e.ID == id {
-			notification = e
-			found = true
-			break
-		}
+func (r *NotificationRepository) Create(notification models.Notification) models.Notification {
+	r.RLock()
```
Comment (ozakernychnyi) [inline internal/app/repositories/notification.go]: it is not good idea to use RLock here, you can lose updated length of your Events slice, so better start this method with
```
r.Lock()
defer r.Unlock()
```

**Review by ozakernychnyi — CHANGES_REQUESTED** [review]

File: internal/app/repositories/notification.go (hunk 7 above)
Comment (kostiantyn-lysenko) [reply internal/app/repositories/notification.go]: ok

**Pull request merged**